include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,129 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#define RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#include <cmath>
+#include <limits>
+#include <type_traits>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/view_facade.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-views
+        /// @{
+
+        /// Distributes `n >= 2` values linearly in the closed interval [`from`, `to`].
+        template<typename T>
+        struct linear_distribute_view
+          : view_facade<linear_distribute_view<T>, finite>
+        {
+            static_assert(std::is_arithmetic<T>(),
+                          "T must model the Arithmetic<T> concept; that is, "
+                          "it must be one of the native integer or floating-point types");
+        private:
+            friend range_access;
+
+            T from_, to_, spacing_;
+
+            constexpr T read() const
+            {
+                return from_;
+            }
+            RANGES_CXX14_CONSTEXPR void next()
+            {
+                from_ += spacing_;
```

Comment:
pinging @sean-parent . The implementation provided supports `RandomAccess`, has no divide in the tight loop, and produces the same results as Matlab's `linspace` function.

I think it is a mistake to enable `RandomAccess` for floating point values (i'll fix that) since `from_ = n * spacing_` and ` while (n > 0) from_ += spacing` won't produce the same result due to floating point errors, but with a constant spacing it can still be a `BidirectionalRange`.